atom/browser/ui/win/jump_list.cc

```diff
@@ -0,0 +1,331 @@
+    if (base::win::SetBooleanValueForPropertyStore(
+        property_store, PKEY_AppUserModel_IsDestListSeparator, true))
+      return SUCCEEDED(collection->AddObject(shell_link));
+  }
+  return false;
+}
+
+bool AppendFile(const JumpListItem& item, IObjectCollection* collection) {
+  DCHECK(collection);
+
+  CComPtr<IShellItem> file;
+  if (SUCCEEDED(SHCreateItemFromParsingName(
+        item.path.value().c_str(), NULL, IID_PPV_ARGS(&file))))
+    return SUCCEEDED(collection->AddObject(file));
+
+  return false;
+}
+
+bool GetShellItemFileName(IShellItem* shell_item, base::FilePath* file_name) {
+  DCHECK(shell_item);
+  DCHECK(file_name);
+
+  wchar_t* file_name_buffer = nullptr;
+  if (SUCCEEDED(shell_item->GetDisplayName(SIGDN_FILESYSPATH,
+                                           &file_name_buffer))) {
+    *file_name = base::FilePath(file_name_buffer);
+    ::CoTaskMemFree(file_name_buffer);
+    return true;
+  }
+  return false;
+}
+
+bool ConvertShellLinkToJumpListItem(IShellLink* shell_link,
+                                    JumpListItem* item) {
+  DCHECK(shell_link);
+  DCHECK(item);
+
+  item->type = JumpListItem::Type::TASK;
+  wchar_t path[MAX_PATH];
+  constexpr size_t max_path = std::extent<decltype(path)>::value;
+  if (FAILED(shell_link->GetPath(path, max_path, nullptr, 0)))
+    return false;
+
+  CComQIPtr<IPropertyStore> property_store = shell_link;
+  base::win::ScopedPropVariant prop;
+  if (SUCCEEDED(property_store->GetValue(PKEY_Link_Arguments, prop.Receive()))
+      && (prop.get().vt == VT_LPWSTR)) {
+    item->arguments = base::string16(prop.get().pwszVal);
+  } else {
+    item->arguments.clear();
+  }
+
+  if (SUCCEEDED(property_store->GetValue(PKEY_Title, prop.Receive()))
+      && (prop.get().vt == VT_LPWSTR)) {
+    item->title = base::string16(prop.get().pwszVal);
+  } else {
+    item->title.clear();
+  }
+
+  int icon_index;
+  if (SUCCEEDED(shell_link->GetIconLocation(path, max_path, &icon_index))) {
+    item->icon_path = base::FilePath(path);
+    item->icon_index = icon_index;
+  } else {
+    item->icon_path.clear();
+    item->icon_index = 0;
+  }
+
+  wchar_t item_desc[INFOTIPSIZE];
+  if (SUCCEEDED(shell_link->GetDescription(
+      item_desc, std::extent<decltype(item_desc)>::value))) {
+    item->description = base::string16(item_desc);
+  } else {
+    item->description.clear();
+  }
+
+  return true;
+}
+
+// Convert IObjectArray of IShellLink & IShellItem to std::vector.
+void ConvertRemovedJumpListItems(IObjectArray* in,
+                                 std::vector<JumpListItem>* out) {
+  DCHECK(in);
+  DCHECK(out);
+
+  UINT removed_count;
+  if (SUCCEEDED(in->GetCount(&removed_count) && (removed_count > 0))) {
+    out->reserve(removed_count);
+    JumpListItem item;
+    IShellItem* shell_item;
+    IShellLink* shell_link;
+    for (UINT i = 0; i < removed_count; ++i) {
+      if (SUCCEEDED(in->GetAt(i, IID_PPV_ARGS(&shell_item)))) {
```

Comment:
The same could be done with `CComPtr` so I don't see the need to use `base::win::ScopedComPtr` here unless you want me to replace all the other `CComPtr`s in this file. I didn't use `CComPtr` in this case because the loop is very simple and constructing two `CComPtr`s every iteration didn't seem like it was worth the effort.